clogMonitor/src/components/TimeRange.js

```diff
@@ -1,5 +1,93 @@
+ * @param {string} props.time - The string representing the current time, format: YYYY-MM-DDTHH:mm:ss
+ * 
+ * @returns {boolean} 
+ * @see {@link https://stackoverflow.com/questions/11887934/how-to-check-if-dst-daylight-saving-time-is-in-effect-and-if-so-the-offset}
+ */
+function hasDSTeffect(time) {
+    const parsedDate = new Date(time)
+    const jan = new Date(parsedDate.getFullYear(), 0, 1); 
+    const jul = new Date(parsedDate.getFullYear(), 6, 1); 
+    const stdOffset = Math.max(jan.getTimezoneOffset(), jul.getTimezoneOffset()); // Standard time offset
+    if(parsedDate.getTimezoneOffset() < stdOffset) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Returns true if chosen date/time has conflict during DST clock changing period
+ * 
+ * @param {Object} props
+ * @param {string} props.time - The string representing the current time, format: YYYY-MM-DDTHH:mm:ss
+ * 
+ * @returns {boolean} 
+ */
+export function hasDSTconflict(time) {
+
+    if (!hasDSTeffect(time)) {
+        return false;
+    }
+    const parsedDate = new Date(time)
+    let curHour = parsedDate.getHours();
+    let curMinute = parsedDate.getMinutes();
+    // Check to see if chosen date/time is the end date/time of DST
+    const curMidnight = new Date(parsedDate.getFullYear(), parsedDate.getMonth(), parsedDate.getDate(), 0).getTimezoneOffset();
+    const cur3AM = new Date(parsedDate.getFullYear(), parsedDate.getMonth(), parsedDate.getDate(), 3).getTimezoneOffset();
+    if (curMidnight < cur3AM && curHour === 1 && curMinute >= 0 && curMinute < 60){
+            return true;
+       }
+    return false;
+}
+
+/**
+ * Returns true if chosen date/time does not exist (e.g. 2021-3-14 2:30AM)
+ * 
+ * @param {Object} props
+ * @param {string} props.time - The string representing the current time, format: YYYY-MM-DDTHH:mm:ss
+ * 
+ * @returns {boolean}
+ */
+export function hasDSTerror(time) {
+    if (!hasDSTeffect(time)) {
+        return false;
+    }
+    const parsedDate = new Date(time)
+    let curHour = parsedDate.getHours();
+    if (curHour!==parseInt(time.substring(11,13))) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Returns converted time to UTC in conflicting DST clock change period 
+ * 
+ * @param {Object} props
+ * @param {string} props.datetimeString - The string representing local date/time, format: YYYY-MM-DDTHH:mm:ss
+ * @param {boolean} hasDST_conflict - The boolean value representing whether chosen date/time has DST conflict
+ * @param {string} choice - The string representing your choice of BEFORE or AFTER
+ *
+ * @returns {string}
+ */
+
+export function convertDSTtoUTC(datetimeString, hasDST_conflict, choice) {
+    const dateTime = new Date(datetimeString);
+    if(hasDST_conflict) {
+        if(choice === "AFTER") {
+            return moment(dateTime).add(1, 'hours');
+        }
+    }
+    return dateTime;
+}
 
 /**
  * Wrapper on two datetime-local inputs representing a start and end datetime respectively.
```

Comment:
You can put yourself as an author too if you want to take credit for all your work on this!